### DISCUSSION TITLE
Can't change default branch anymore from the web?

### DISCUSSION THREAD

**Discussion by seanmavley** [community/community#120083] [Repositories] [answered]
### Select Topic Area

Question

### Body

Where's the set default option?

![image](https://github.com/community/community/assets/5289083/02925a44-832b-4fe5-813c-30e7a09bf562)

According to this article: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch the default branch should have the option there.

Where is it?

**Comment by queenofcorgis** [comment]
Hi @seanmavley 👋🏼 

Do you see the option to switch the default branch when you click the three dots for the `main-old` branch? That one looks like it is currently the default which is the branch you'd see this option on, not the one you want to switch to ❤️

**Comment by seanmavley** [comment, marked as answer]
cc @queenofcorgis 

> That one looks like it is currently the default which is the branch you'd see this option on, not the one you want to switch to ❤️

Nope. I only get 3 options as per image below, and changing default branch isn't part of it.

![image](https://github.com/community/community/assets/5289083/6ddf204f-9f4b-43e7-bb18-a8ae31d061f7)

If you guys have removed the option from the web, as in option to swap default branch, no problem, just say so. It's just not there on the web anymore.

As they say, `perfection is achieved not when there is nothing more to add but when there is nothing left to take away` - Keep taking away things from the web. *sigh*

# Answer:

By the way, I updated the default branch via github cli with

`gh repo edit <username>/<repo> --default-branch main`

Edit:

Ha!

So, change of default is located `https://github.com/<username>/<repo>/settings`
Right. Can't change default branch on branch list page, rather on the repo settings page.

**Reply by zombiecat69** [reply]
This was very useful feature and suddenly I see it gone! :( Thanks for pasting here how to do it with cli! @seanmavley

**Reply by ubeogesh** [reply]
this answer doesn't work anymore. 

```
C:\Users\myuser>gh repo edit MyCompany/my-repo
? What do you want to edit? Default Branch Name
? Default branch name (develop) master                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              ? Default branch name master
HTTP 422: Validation Failed (https://api.github.com/repos/MyCompany/my-repo)
Repository.default_branch is invalid
```
